src/config/active-response.c

```diff
@@ -428,10 +593,206 @@ int ReadActiveCommands(XML_NODE node, void *d1, __attribute__((unused)) void *d2
+        }
         free(tmp_command);
         return (-1);
     }
 
+    if (output) {
+        if (tmp_command->name) {
+            free (tmp_command->name);
+        }
+        
+        if (tmp_command->executable) {
+            free (tmp_command->executable);
+        }
+
+        if (tmp_command->extra_args) {
+            free (tmp_command->extra_args);
+        }
+
+        if (tmp_command) {
+            free (tmp_command);
+        }
+
+    }
+
     return (0);
 }
+
+int Test_ActiveResponse(const char *path, int type, char **output) {
+    int fail = 0;
+    OSList *test_activecmd;
+    OSList *test_activeresp;
+
+    test_activecmd = OSList_Create();
+    test_activeresp = OSList_Create();
+
+    if (!test_activecmd || !test_activeresp) {
+        if (output == NULL){
+            merror(LIST_ERROR, "Attempting to check Active-Response");
+        } else {
+            wm_strcat(output, "Active-Response: Unable to create a new list", '\n');
+        }      
+        return OS_INVALID;
+    }
+
+    /* type indicates local or remote config */
+    if (ReadConfig(CAR | type, path, test_activecmd, test_activeresp, output) < 0) {
+        if (output == NULL){
+            merror(CONF_READ_ERROR, "Active-Response");
+        } else {
+            wm_strcat(output, "Invalid configuration in Active-Response", '\n');
+        }
+        fail = 1;
+    }
+
+    /* Frees the LogReader config struct */
+    Free_OSList(test_activecmd);
+    Free_OSList(test_activeresp);
+
+    os_free (test_activecmd);
+    os_free (test_activeresp);
+
+    if (fail) {
+        return OS_INVALID;
+    }
+
+    return 0;
+}
+
+int Test_Agent_Active_Response(const char *path, char **output) {
+    const char *(xmlf[]) = {"ossec_config", "active-response", "disabled", NULL};
+    const char *(castore[]) = {"ossec_config", "active-response", "ca_store", NULL};
+    const char *(caverify[]) = {"ossec_config", "active-response", "ca_verification", NULL};
+
+    /* Read XML file */
+
+    OS_XML xml;
+    if (OS_ReadXML(path, &xml) < 0) {
+        if (output == NULL){
+            merror(XML_ERROR, path, xml.err, xml.err_line);
+        } else {
+            wm_strcat(output, "ERROR: Invalid configuration in Active-Response", '\n');
+        }
+        return OS_INVALID;
+    }
+
+    char *disable_entry = OS_GetOneContentforElement(&xml, xmlf);
+    if (disable_entry) {
+        if ( strcmp(disable_entry, "yes") && strcmp(disable_entry, "no") ) {
+            if (output == NULL){
+                merror(XML_VALUEERR, "disabled", disable_entry);
+            } else {
+                wm_strcat(output, "ERROR: Invalid configuration in Active-Response", '\n');
+            }
+            os_free(disable_entry);
+            OS_ClearXML(&xml);
+            return OS_INVALID;
+        }
+
+        os_free(disable_entry);
+    }
+
+    XML_NODE node = OS_GetElementsbyNode(&xml, NULL);
+    XML_NODE child = NULL;
+    char *repeated_t = NULL;
```

Comment:
*_t usually refer to typedefs, I encourage strongly for repeated_t to be renamed into repeated os something more meaningful.

Reply:
Also, repeated_a (declared below) should be renamed into something more meaningful to improve code readability.